aten/src/ATen/native/cuda/ForeachBinaryOpScalarList.cu

```diff
@@ -6,18 +6,23 @@
 namespace at { namespace native {
 
 template<template<class> class Op>
-std::vector<Tensor> foreach_binary_op(TensorList tensors, at::ArrayRef<Scalar> scalars) {
+std::vector<Tensor> foreach_binary_op(TensorList tensors, at::ArrayRef<Scalar> scalars, bool promote_integer_float=false) {
     std::vector<std::vector<at::Tensor>> tensor_lists;
+    std::vector<at::Tensor> temp_tensors;
+    temp_tensors.reserve(tensors.size());
+    tensor_lists.emplace_back(temp_tensors);
     std::vector<at::Tensor> vec_res;
     vec_res.reserve(tensors.size());
-    for (const auto& t: tensors) {
-        vec_res.emplace_back(at::native::empty_like(t));
+
+    ScalarType result_type = get_result_type(tensors[0], scalars[0], promote_integer_float);
+    for (int i = 0; i < tensors.size(); i++) {
+        tensor_lists[0].emplace_back(tensors[i].to(result_type));
```

Comment:
lest I forget, this is not a good way to do type promotion - it will launch as many kernels as "slow" path (each `.to` launches a kernel), and will probably result in more memory traffic.